pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -7,216 +7,739 @@
+         print('File not found.')
+         raise e
+    # load gradient of all constraints (sparse)
+    # If no constraint exists, return []
+    num_constraints = len(list(model.component_data_objects(Constraint,
+                                                            active=True,
+                                                            descend_into=True)))
+    if num_constraints > 0 :
+        try:
+            gradient_c = np.loadtxt("./GJH/A_print.txt")
+        except Exception as e:
+            print('./GJH/A_print.txt not found.')
+        row_idx = gradient_c[:,1]-1
+        col_idx = gradient_c[:,0]-1
+        data = gradient_c[:,2]
+        gradient_c = sparse.csr_matrix((data, (row_idx, col_idx)), shape=(len(row)-1, len(col)))
     else:
-        m = instance
+        gradient_c = np.array([])
+    # remove all generated files
     
-    #Generate component maps for associating Variables to perturbations
-    varSubList = []
-    for parameter in paramSubList:
-        tempName = unique_component_name(b,parameter.local_name)
-        b.add_component(tempName,Var(parameter.index_set()))
-        myVar = b.component(tempName)
-        varSubList.append(myVar)
- 
-    #Note: substitutions are not currently compatible with 
-    #      ComponentMap [ECSA 2018/11/23], this relates to Issue #755
-    paramCompMap = ComponentMap(zip(paramSubList, varSubList))
-    variableSubMap = {}
-    #variableSubMap = ComponentMap()
-    paramPerturbMap = ComponentMap(zip(paramSubList,perturbList))
-    perturbSubMap = {}
-    #perturbSubMap = ComponentMap()
-   
-    paramDataList = [] 
-    for parameter in paramSubList:
-        # Loop over each ParamData in the Param Component
-        #
-        # Note: Sets are unordered in Pyomo.  For this to be
-        # deterministic, we need to sort the index (otherwise, the
-        # ordering of things in the paramDataList may change).  We use
-        # sorted_robust to guard against mixed-type Sets in Python 3.x
-        for kk in sorted_robust(parameter):
-            variableSubMap[id(parameter[kk])]=paramCompMap[parameter][kk]
-            perturbSubMap[id(parameter[kk])]=paramPerturbMap[parameter][kk]
-            paramDataList.append(parameter[kk])
-
-    #clone Objective, add to Block, and update any Expressions
-    for cc in list(m.component_data_objects(Objective,
-                                            active=True,
-                                            descend_into=True)):
-        tempName=unique_component_name(m,cc.local_name)    
-        b.add_component(tempName,
-                  Objective(expr=ExpressionReplacementVisitor(
-                  substitute=variableSubMap,
-                  remove_named_expressions=True).dfs_postorder_stack(cc.expr)))
-        cc.deactivate()
+    shutil.move("col_row.nl", "./GJH/")
+    shutil.move("col_row.col", "./GJH/")
+    shutil.move("col_row.row", "./GJH/")
+    shutil.rmtree('GJH', ignore_errors=True)
```

Comment:
It would be much better to parse this data immediately and remove the files, instead of polluting the user's current working directory (which may not even be writable) with hard-coded file names,